packages/nodes-base/nodes/Webhook/Webhook.node.ts

```diff
@@ -125,6 +125,17 @@ export class Webhook extends Node {
 			return this.handleFormData(context);
 		}
 
+		const nodeVersion = context.getNode().typeVersion;
+		if (nodeVersion > 1 && !req.body && !options.rawBody) {
+			try {
+				return await this.handleBinaryData(context);
+			} catch (error) {}
+		}
```

Comment:
This change means that now every single `GET` webhook gets a `.binary` object set, even though there is not binary data present. 
This causes errors [here](https://github.com/n8n-io/n8n/blob/master/packages/cli/src/executionLifecycleHooks/restoreBinaryDataId.ts#L23).
We need to 
1. improve the checks here to create `.binary` if there actually any data (or maybe just check if the method is `POST`, `PATCH`, or `PUT).
2. update the `workflowExecuteAfter` hook to make sure that `restoreBinaryDataId` does not fail.

Reply:
> improve the checks here to create .binary if there actually any data

it's checked in `handleBinaryData`, agree that we could add check for `httpMethod` type as generally only those methods would have binaries

```
			const binaryPropertyName = (options.binaryPropertyName || 'data') as string;
			const fileName = req.contentDisposition?.filename ?? uuid();
			const binaryData = await context.nodeHelpers.copyBinaryFile(
				binaryFile.path,
				fileName,
				req.contentType ?? 'application/octet-stream',
			);

			if (!binaryData.data) {
				return { workflowData: [[returnItem]] };
			}

			returnItem.binary![binaryPropertyName] = binaryData;
```